packages-web/datagrid/src/Datagrid.editorPreview.tsx

```diff
@@ -0,0 +1,83 @@
+import { createElement, ReactElement, useMemo } from "react";
+import { DatagridPreviewProps } from "../typings/DatagridProps";
+
+import "./ui/Datagrid.scss";
+import { Table } from "./components/Table";
+
+export function preview(props: DatagridPreviewProps): ReactElement {
+    const {
+        class: className,
+        headerWidgets,
+        footerWidgets,
+        pagingPosition,
+        columnsSortable,
+        columnsFilterable,
+        columnsResizable,
+        columnsDraggable,
+        columnsHidable,
+        columns: columnProps,
+        pagingEnabled,
+        pageSize
+    } = props;
+
+    const columns = columnProps.map((column, index) => ({
+        Header: column.header,
+        accessor: `col_${index}`,
+        filter: "text"
+    }));
+
+    const columnsConfig = columnProps
+        .map((column, index) => {
+            const { ...data } = column;
+            delete data.content;
+            delete data.attribute;
+            return {
+                [`col_${index}`]: data
+            };
+        })
+        .reduce((acc, current) => ({ ...acc, ...current }), {});
+
+    const data = useMemo(
+        () =>
+            [1, 2, 3, 4, 5].map(() =>
+                columnProps
+                    .map((column, index) => ({
+                        [`col_${index}`]: column.attribute,
+                        [`col_${index}_hasWidgets`]: false,
+                        [`content_col_${index}`]:
+                            column.content && column.content.widgetCount > 0 ? (
```

Comment:
We have to render renderer unconditionally, even when there are no `widgetCount`, since there should be a way to add a first widget.

Reply:
I've fixed this in previous comments